backend/src/main/java/gov/cdc/usds/simplereport/config/FileValidatorConfiguration.java

```diff
@@ -0,0 +1,20 @@
+package gov.cdc.usds.simplereport.config;
+
+import gov.cdc.usds.simplereport.validators.FileValidator;
+import gov.cdc.usds.simplereport.validators.PatientUploadRow;
+import gov.cdc.usds.simplereport.validators.TestResultRow;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
```

Comment:
This is neat, I haven't seen this configuration style before! I'm curious, what's the usage if we're manually declaring `FileValidator<type>` where it's being used?

Reply:
I'm not sure if this answers your question, but these are actually being used to autowire the validator in the services.  The only place we are declaring file validator is in the tests.
![image](https://user-images.githubusercontent.com/10108172/203416875-db0fc39d-db9b-4050-affd-2b1b78c3111e.png)
![image](https://user-images.githubusercontent.com/10108172/203416921-84877794-1d15-43bb-8ea6-1c5cb56a467b.png)

If the question was around why we are passing around `TestResultRow::new` into the functions, it's because java can't instantiate generic types for example in FileValidator, you cannot do `var patientUploadRow = new T()`. After doing a little more research it seems like another way to do accomplish creating the generic is by accepting a class into the constructor and then creating it with that class reference, but that way requires a lot of exception handling. I do think this way may look a little nicer passing in a class, but I'd prefer not having to worry about any error handling. 
![image](https://user-images.githubusercontent.com/10108172/203420444-846c0ef5-1d28-46ac-a764-c51170cf6894.png)
![image](https://user-images.githubusercontent.com/10108172/203419456-4a7d099c-e023-4877-8a93-24abbf9aaeb9.png)
![image](https://user-images.githubusercontent.com/10108172/203419336-a6e3c1da-24e4-4420-bd5b-53d2c59a19c1.png)

References: https://stackoverflow.com/questions/2434041/instantiating-generics-type-in-java

Reply:
Ahhhh right, the `Autowire` use case makes a lot of sense. That's mostly what I was asking about - why it had to be a config bean vs. just another POJO. 

I agree with you on passing the class vs new - I wouldn't want to do the exception handling either, and I think the `TestResultRow::new` is still perfectly readable.

Thanks for the detailed explanation!